File: JANL.VB/SQL/SQLProcedure.vb
Namespace SQL
	''' <summary>
	''' Процедура возвращающая <typeparamref name="T"/>
	''' </summary>
	''' <typeparam name="T">Тип значения возвращаемого процедурой</typeparam>
	Public MustInherit Class SQLProcedure(Of T)
		Inherits BaseSQLCommand(Of T)

		''' <summary>
		''' Создаёт новую процедуру с именем вызывающего метода
		''' </summary>
		Public Sub New(Name As String)
			MyBase.New(Name, CommandType.StoredProcedure)
		End Sub

	End Class
End Namespace

File: JANL.CS/SQL/SQLProcedure.cs
using System.Data;

namespace JANL.SQL
{   /// <summary>
	/// Процедура возвращающая <typeparamref name="T"/>
	/// </summary>
	/// <typeparam name="T">Тип значения возвращаемого процедурой</typeparam>
	public abstract class SQLProcedure<T> : BaseSQLCommand<T>
	{
		/// <summary>
		/// Создаёт новую процедуру с именем вызывающего метода
		/// </summary>
		public SQLProcedure(string Name) : base(Name, CommandType.StoredProcedure) { }
	}
}